src/sentry/static/sentry/app/views/organizationGroupDetails/reprocessingDialogForm.tsx

```diff
@@ -53,46 +64,71 @@ function ReprocessingDialogForm({group, organization, Header, Body, closeModal}:
     browserHistory.push(
       `/organizations/${orgSlug}/issues/?query=reprocessing.original_issue_id:${group.id}/`
     );
+  };
+
+  handleError() {
+    addErrorMessage(t('Failed to reprocess. Please check your input.'));
   }
 
-  return (
-    <React.Fragment>
-      <Header closeButton>{title}</Header>
-      <Body>
-        <Introduction>
-          {t(
-            'Reprocessing applies any new debug files or grouping configuration to an Issue. Before you give it a try, you should probably consider these impacts:'
-          )}
-        </Introduction>
-        <StyledList symbol="bullet">
-          {impacts.map((impact, index) => (
-            <ListItem key={index}>{impact}</ListItem>
-          ))}
-        </StyledList>
-        <ApiForm
-          apiEndpoint={endpoint}
-          apiMethod="POST"
-          footerClass="modal-footer"
-          onSubmitSuccess={handleSuccess}
-          submitLabel={title}
-          submitLoadingMessage={t('Reprocessing\u2026')}
-          submitErrorMessage={t('Failed to reprocess. Please check your input.')}
-          onCancel={closeModal}
-        >
-          <NumberField
-            name="maxEvents"
-            label={t('Enter the number of events to be reprocessed')}
-            help={tct(
-              'You can limit the number of events reprocessed in this Issue. If you set a limit, we will reprocess your most recent events, [strong:and the rest will be deleted.]',
-              {strong: <strong />}
+  // typing of args is based on
+  // https://github.com/getsentry/sentry/blob/5a865648d9ced8c773539b718aefa2eb155995ab/src/sentry/static/sentry/app/components/forms/numberField.tsx#L17-L28
+  handleMaxEventsChange = (maxEvents: number | '') => {
```

Comment:
I checked this locally and `maxEvents` it's always of type string

Reply:
coerceValue is called which I linked in code comments, then onChange is called: https://github.com/getsentry/sentry/blob/26aa2e891bdd56300e9b9d1c56a136a545029978/src/sentry/static/sentry/app/components/forms/formField.tsx#L140-L141

Reply:
the `numberField` that makes use of the `coerciveValue` func is not the one you are using, but this:
https://github.com/getsentry/sentry/blob/master/src/sentry/static/sentry/app/components/forms/numberField.tsx